src/rgw/rgw_auth.h

```diff
@@ -765,7 +765,8 @@ class LocalApplier : public IdentityApplier {
                                       std::vector<IAM::Policy> policies,
                                       const std::string& subuser,
                                       const std::optional<uint32_t>& perm_mask,
-                                      const std::string& access_key_id) const = 0;
+                                      const std::string& access_key_id,
+                                      bool is_impersonating = false) const = 0;
```

Comment:
as per https://google.github.io/styleguide/cppguide.html#Default_Arguments, please avoid default arguments in virtual functions. the compiler won't check that all overrides use the same default